canvas_site_creator/api.py

```diff
@@ -0,0 +1,402 @@
+import json
+import logging
+
+from django.conf import settings
+from django.contrib.auth.decorators import login_required
+from django.db.models import Q
+from django.http import JsonResponse
+from django.utils import timezone
+from django.views.decorators.http import require_http_methods
+from canvas_course_site_wizard.models import  CanvasCourseGenerationJob
+from canvas_sdk.exceptions import CanvasAPIError
+from canvas_sdk.methods.content_migrations import create_content_migration_courses
+from canvas_sdk.methods.courses import create_new_course
+from canvas_sdk.methods.sections import create_course_section
+from icommons_common.canvas_api.helpers import accounts as canvas_api_accounts
+from icommons_common.canvas_utils import SessionInactivityExpirationRC
+from icommons_common.models import CourseInstance, Person
+from icommons_common.view_utils import create_json_200_response, \
+    create_json_500_response
+from lti_permissions.decorators import lti_permission_required
+
+from .models import (
+    get_course_instance_query_set,
+    get_course_instance_summary_data,
+    get_course_job_summary_data
+)
+from .utils import (
+    get_school_data_for_sis_account_id,
+    get_term_data_for_school,
+    get_department_data_for_school,
+    get_course_group_data_for_school,
+)
+
+logger = logging.getLogger(__name__)
+
+COURSE_INSTANCE_DATA_FIELDS = ('course_instance_id', 'course_instance_id',
+                               'course__registrar_code_display','title',
+                               'section')
+BULK_JOB_DATA_FIELDS = ('created_at', 'status')
+COURSE_JOB_DATA_FIELDS = ('created_at', 'workflow_state')
+
+SDK_CONTEXT = SessionInactivityExpirationRC(**settings.CANVAS_SDK_SETTINGS)
+
+def _unpack_datatables_params(request):
+    return (
+        int(request.GET.get('draw', 0)),
+        int(request.GET.get('start', 0)),
+        int(request.GET.get('length', 10)),
+        int(request.GET.get('order[0][column]', 3)),
+        request.GET.get('order[0][dir]', 'asc'),
+        request.GET.get('search[value]')
+    )
+
+
+@login_required
+@lti_permission_required(canvas_api_accounts.ACCOUNT_PERMISSION_MANAGE_COURSES)
```

Comment:
Do we want to make these permissions a little more local/based on what's provided in this project's `settings/base.py`?